returnn/tf/util/basic.py

```diff
@@ -929,13 +929,14 @@ def get_tf_list_local_devices(tf_session_opts=None):
   You can get the list available in a given TF session by :func:`tf.compat.v1.Session.list_devices`.
 
   :param dict[str]|None tf_session_opts: if given, will init a temp tf.compat.v1.Session with these opts
+  :param io.FileIO file: file stream for print statements, defaults to sys.stdout
```

Comment:
I think the type is wrong. It should be `io.TextIOBase`. Or maybe even more general, like `io.TextIOBase|io.StringIO|typing.TextIO|None`.

Reply:
I assumed the existing type had a reason (in `print_available_devices`). I will also correct that one as well.

Reply:
It's a bit inconsistent and also wrong in other places. You will not always get warnings, but you will get warnings when you use different types. E.g. what should work (without warning) is `sys.stdout`, result from `open()`, and also `StringIO()`.